### PR TITLE
[LTD-3401] Bugfix queue view tab selection page 2

### PR DESCRIPTION
**Pull request by currycoder**
### Aim

This change achieves two things;
- Ensure that the current `page` reset when clicking a tab ("Cases to review"/"My cases"/"Open queries") on the queue view page.
- Ensure that the application returns a 404 gracefully when a user navigates to a queue view `page` for which the API does not have results.

https://uktrade.atlassian.net/browse/LTD-3401

### CODE REVIEW

File: caseworker/queues/views.py [hunk 1]
```diff
@@ -67,8 +67,14 @@ def queue_pk(self):
...
         params = self.get_params()
 
-        data = get_cases_search_data(self.request, self.queue_pk, params)
-        return data
+        response = get_cases_search_data(self.request, self.queue_pk, params)
+        if not response.ok:
+            if response.status_code == 404:
+                raise Http404()
+            else:
+                raise HttpResponseServerError()
```
Comment (currycoder) [inline caseworker/queues/views.py:75]: Discussed with Kevin offline; make this raise a `ServiceError` instead https://github.com/uktrade/lite-frontend/blob/225c60d4232e21d6adbc3d69f31a1464f4cffb75/core/exceptions.py#L5
Also ensure that this line is tested.

**Review by Harriethw — APPROVED** [review]
👍

File: unit_tests/caseworker/queues/test_views.py [hunk 2]
```diff
@@ -380,6 +394,26 @@ def test_cases_home_page_nca_applicable_search(authorized_client, mock_cases_sea
...
+    url = reverse("queues:cases")
+    response = authorized_client.get(url)
+    assert response.status_code == 404
+
+
+def test_cases_home_page_case_search_API_error(authorized_client, mock_cases_search_error):
+    url = reverse("queues:cases")
+    authorized_client.raise_request_exception = True
+    error = None
+    try:
```
Comment (kevincarrogan) [inline unit_tests/caseworker/queues/test_views.py:407]: `with pytest.raises` could be used here instead.

Comment (kevincarrogan) [reply unit_tests/caseworker/queues/test_views.py:407]: https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

**Review by kevincarrogan — APPROVED** [review]

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**